### DISCUSSION TITLE
Iframe https Mixed content

### DISCUSSION THREAD

**Discussion by corentin-begne** [kimai/kimai#5003] [On-Premise: Installation & Updates]
### Describe the issue

I am trying to integrate Kimai within an iFrame through the external application feature in Nextcloud. However, I am encountering a mixed content error. Kimai seems to be making HTTP requests even though the site is served over HTTPS, which results in the mixed content warning in the browser.
I have found this thread that discusses a similar issue related to reverse proxies: [Forum Thread](https://forum.cloudron.io/topic/4848/mixed-content-warning-while-embedding), but my setup does not involve a reverse proxy because I use the cloned repo directly with Apache in Ubuntu 22.04 not a reverse proxy with the image inside docker.
Anyway you can see the same behavior outside the iframe, you've got a double access, http which redirect to https.
If I logging outside iframe after it works but it seems the redirection to the login page and all login requests makes this isssue.
Can you help me please ?
Thx !

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.13.0

### How do you run Kimai?

Other (please mention below)

### Which PHP version are you using?

8.2

**Comment by kevinpapst** [comment]
This is an issue on your end.

**Comment by kevinpapst** [comment]
Either an invalid redirect or webserver config.
Combined with a aggressive browser cache. Use private mode.

**Comment by corentin-begne** [comment]
I have the problem even in private mode.
I use the apache config provided by kimai documentation
https://www.kimai.org/documentation/webserver-configuration.html
Why the issue occur only on the login redirection ?
I tried many things but nothing working I always have the first redirection in http 200 Temporary Redirect.
What could I do ?

**Comment by corentin-begne** [comment]
Do you have any clues of what could do that on my side ?
As I see it's like there's an http redirection hardcoded on your side on the redirections, I try many config in Symfony and server configuration but noting working.

**Comment by kevinpapst** [comment]
> As I see it's like there's an http redirection hardcoded on your side on the redirections

There is no "my side".
There is no "hard-coded" redirect.
If you request http, there is a redirect to http. That is exactly as it should be.
If your reverse proxy is not correctly forwarding the necessary headers then the protocol might be mis-identified.
I am pretty sure this is an issue in the virtual host config of your reverse proxy.

**Comment by corentin-begne** [comment]
But I d'ont use a reverse proxy, kimai is not in docker and I try to access to the root kimai url in https which redirect in http to the  login then https, same for 2fa even if I try to set the login url in https directly instead.
How it could be on my side ?
Sharing my vhost, same result whithout Header instructions.
```bash
<VirtualHost *:443>
        ServerAlias xxx.com
        DocumentRoot /var/www/kimai/public/
        Header always unset X-Frame-Options
        Header set Content-Security-Policy "frame-ancestors 'self' https://xxx.com"
        Header always edit Set-Cookie (.*) "$1; SameSite=None; Secure"

        <Directory /var/www/kimai/public/>
                AllowOverride All
                Require all granted
                FallbackResource /index.php
        </Directory>

        <Directory /var/www/kimai>
                Options FollowSymlinks
        </Directory>

        <Directory /var/www/kimai/public/bundles>
                FallbackResource disabled
        </Directory>

        ErrorLog /var/log/apache2/kimai_error.log
        CustomLog /var/log/apache2/kimai_access.log combined
</VirtualHost>
```

**Reply by kevinpapst** [reply]
There is a .htaccess in public/
If you roll your own, then you have to take care that they don't interfer.

**Comment by corentin-begne** [comment]
I didn't modify the .htaccess

**Reply by kevinpapst** [reply]
Paid support is always available at supprot@kimai.org

**Reply by corentin-begne** [reply]
> Paid support is always available at [supprot@kimai.org](mailto:supprot@kimai.org)

Yes but I won't pay for support just for this, its just to bad for me when all my others sites can be integrated inside Nextcloud without any problem (using or not a reverse proxy) I have to let kimai be accessible outside, the only one.
Thank you anyway for your time.

**Reply by Rudiberto** [reply]
> Paid support is always available at [supprot@kimai.org](mailto:supprot@kimai.org)

For sure, it's "support" not "supprot@kimai.org"
;-)
